numpy/lib/format.py

```diff
@@ -259,6 +259,33 @@ def dtype_to_descr(dtype):
     else:
         return dtype.str
 
+def descr_to_dtype(descr):
+    if isinstance(descr, str):
+        # descr was produced by dtype.str, so this always works
+        return numpy.dtype(descr)
+
+    fields = []
+    offset = 0
+    for field in descr:
+        if len(field) == 2:
+            name, descr_str = field
+            dt = descr_to_dtype(descr_str)
+        else:
+            name, descr_str, shape = field
+            dt = numpy.dtype((descr_to_dtype(descr_str), shape))
+
+        # ignore padding bytes, which will be void bytes with '' as name
+        # (once blank fieldnames are deprecated, only "if name == ''" needed)
```

Comment:
Do you mean "fieldnames are deprecated" or "support for blank field names is removed"?

Reply:
reworded